### ISSUE TITLE
Rescuezilla for multiple Linux

### ISSUE THREAD

**Issue by RasheedAZ** [rescuezilla/rescuezilla#252]
Hi everyone.

We have 16 identical PCs in our school's computer lab. I'd like to install Linux on them all. One is already running Pop!_OS and I'd like to clone the installation for deployment on the rest of the machines with the same configuration, including the exact setup for Firefox and Cosmic desktop environment.

Would Rescuezilla be the correct tool for the job?

If yes, then is there a guide I could follow please?

I really do not want to install and configure the remaining 15 machines one at a time... Would appreciate any useful advice.

Thanks.

**Comment by shasheene** [comment]
Yes, Rescuezilla would work (simply 'backup' from the source machine, and 'restore' on each of the machines). If any of the hard drives are smaller than the original, [you should use GParted Partition Editor to shrink your first machine before creating a backup](https://github.com/rescuezilla/rescuezilla/wiki/HOWTO:-Restoring-to-a-smaller-disk.-Eg,-1000GB-HDD-to-500GB-SSD). You should be able to use a single USB stick (using 'Load Into RAM' boot menu option), and point to a network shared drive to hose the image.

So the question is do you expect to image your 16 network-connected computers with any regularity? If so Rescuezilla is definitely _not_ the most time-efficient tool, because you would need to manually run Rescuezilla on each machine.

But there is a very powerful tool called [FOG Project](https://fogproject.org/) which can deploy an image to all 70 machines with 1 click based on a network-boot architecture. The big downside with FOG Project is **the initial configuration is requires an IT administrator**. It's a Linux tool that the official instructions recommends is installed in a virtual machine, but most crucially **it needs "DHCP Option 66 and 67" configured on the network's DHCP server**. This option is usually not available on a home network router, but should be available on a school's IT network infrastructure.

I was able to configure FOG Project using [this YouTube playlist](https://www.youtube.com/watch?v=LMuwkJsjlOo&list=PLRR0y1Ha_dRclMpyhs51ZcruocffNR3su) and lots of researching to find the DHCP Option 66 and 67 for my router.

**If you're going to be deploying images onto a large amount of computers with any regularity then upfront effort for the IT system administrator configuring FOG Project is definitely worthwhile**: once FOG Project is installed and configured it's pretty easy to use. Being able to image tens, hundreds (or even thousands) of computers in 1 click is very time efficient. Also FOG Project automatically handles the resizing step, so no need to use GParted Partition Editor here.

**Comment by RasheedAZ** [comment]
Thanks for the detailed reply Shasheen.


> You should be able to use a single USB stick (using 'Load Into RAM' boot menu option), and point to a network shared drive to hose the image.
> 

All of the computers have the same hardware so this step would not be needed, right?

> So the question is do you expect to image your 16 network-connected computers with any regularity? If so Rescuezilla is definitely _not_ the most time-efficient tool, because you would need to manually run Rescuezilla on each machine.
> 

Sorry, not sure what network-connected means. They are all connected to the same WiFi network but to each other. And I only expect to do this once for the initial deployment and maybe every 2-3 years after upgrading to new LTS version.

**Comment by shasheene** [comment]
> Sorry, not sure what network-connected means.

Yes by network-connected I mean connected to the same eg, WiFi network, so that you can open a Terminal and type eg, `ping `192.168.1.10` and get a response. From the sounds of it, the very advanced FOG Project will probably be way too complicated to configure. You only have 16 computers to image, so doing it manually with Rescuezilla might be the simplest and most time efficient solution for you.

> > You should be able to use a single USB stick (using 'Load Into RAM' boot menu option), and point to a network shared drive _*_**that is hosting the image**.
>
> All of the computers have the same hardware so this step would not be needed, right?

Sorry my wording was clumsy, I have modified my original post. What I mean is with Rescuezilla you create a live USB stick that you use to boot on each machine. There is a boot menu option titled "Load Into RAM" which allows you to reuse a single USB stick across all 16 machines. Simply plug it in and once booted to the desktop, unplug the stick. You may find the Rescuezilla v2.2 version labelled 'Focal' always reliably reaches the Desktop where the image labelled 'Hirsute' may not.

If you have computer hosting a network shared folder on the same network (trivially simple to create on a Windows computer by right clicking a folder and from the properties menu changing the share settings), then you can save your Rescuezilla backup to that machine. Then on each of the computers you can point them to the network shared folder and authenticate. This may be easier than carrying around a large external hard drive and plugging it in for each computer. And by using a network shared folder you can restore the image to multiple computers simultaneously. But if you have eg, a 128GB USB stick in addition to your Rescuezilla live USB you can use those two sticks (not as elegant as the network shared folder solution though).

The [GParted Partition Editor stuff](https://github.com/rescuezilla/rescuezilla/wiki/HOWTO:-Restoring-to-a-smaller-disk.-Eg,-1000GB-HDD-to-500GB-SSD) I talked about in the previous comment matters if there is a machine with a smaller hard drive than the source image. If all the hard drives are identical size you don't need to do any resizing using GParted. If some hard drives are larger you'll need to use GParted Partition Editor to increase the final partition so it can completely utilize the hard drive's actual capacity.

**Comment by RasheedAZ** [comment]
> If you have computer hosting a network shared folder on the same network (trivially simple to create on a Windows computer by right clicking a folder and from the properties menu changing the share settings), then you can save your Rescuezilla backup to that machine. Then on each of the computers you can point them to the network shared folder and authenticate. This may be easier than carrying around a large external hard drive and plugging it in for each computer. And by using a network shared folder you can restore the image to multiple computers simultaneously. 

OK. So basically, if I have a shared public folder on a Windows 7 PC connected to the same network, then at this step:

![image](https://user-images.githubusercontent.com/16827461/128601056-8eaca12f-cb9a-41c8-81a0-402d53adb42a.png)

I can just put "//192.168.0.xx/nameofsharedpublicfolder" in the Server or share location field, right?

And I get that IP address with the ipconfig command right?

If I'm understanding this correctly, this might be the best option...

**Comment by shasheene** [comment]
Yes, that should work.

Newer versions of Rescuezilla have the version field specified as a dropdown menu. You may need to have the version field set away from the default value if it doesn't connect the first time, but the value you need to select is pretty self explanatory when you see the version field's dropdown menu.
